docs/secret-template.md

```diff
@@ -1,56 +1,124 @@
+kind: SecretTemplate
+metadata:
+  name: pod-input
+spec:
+  #! list of resources to read information off
+  inputResources:
+  - name: username-secret
+    ref:
+      apiVersion: v1
+      kind: Secret
+      name: username
+  - name: password-secret
+    ref:
+      apiVersion: v1
+      kind: Secret
+      name: password
+  #! the template that follows a subset of the Secret API
+  template:
+    #! data is used for templating in data that *is* base64 encoded, most like Secrets.
+    data:
+      password: $(.password-secret.data.password)
+      username: $(.username-secret.data.username)
+```
+
+Above configuration results in a `helm-postgres` Secret created within `default` namespace:
+
+```console
+Namespace  Name           Kind          Owner    Conds.  Rs  Ri  Age
+default    helm-postgres  Secret        cluster  -       ok  -   1d
 ```
 
-With custom secret projection:
+### SecretTemplate
+
+SecretTemplate CRD allows to template out a Secret from information on other APIs.
+
+`metadata` fields:
+
+- `name`: (required; string) Secret by the same name (in the namespace) will be created.
+
+`spec` fields:
+
+- `serviceAccountName` (required; string) Name of the service account used to read the input resources. If not provided, only Secrets can be read on the `.spec.inputResources`.
+- `inputResources` (required; array of objects) Array of named Kubernetes API resources to read information off.  The name of an input resource can dynamically reference previous input resources by a JSONPath expression, signified by an opening "$(" and a closing ")".
+- `template` (optional; subset of Secret API object) A template of the Secret to be created.  Any string value in the subset can reference information off a resource in `.spec.inputResources` using a JSONPath expression, signified by an opening "$(" and a closing ")".
+
+### Further Example
 
 ```yaml
-apiVersion: secretgen.k14s.io/v1alpha1
-kind: Password
+#! reads the resources created by an instance of the bitnami helm chart https://github.com/bitnami/charts/tree/master/bitnami/postgresql/ and creates a binding secret https://github.com/servicebinding/spec
+---
+apiVersion: secretgen.carvel.dev/v1alpha1
+kind: SecretTemplate
 metadata:
-  name: pg-password
+  name: helm-postgres
 spec:
-  secretTemplate:
-    type: Opaque
+  #! service account with permissions to get/list/watch pods, services, secrets
+  serviceAccountName: helm-reader
+  #! list of resources to read off, these resources can be dynamically specified based on the fields of previously stated resources
+  inputResources:
+  - name: pod
+    ref:
+      apiVersion: v1
+      kind: Pod
+      name: postgres-postgresql-0
+  - name: service
+    ref:
+      apiVersion: v1
+      kind: Service
+      name: postgres-postgresql
+  - name: secret
+    ref:
+      apiVersion: v1
+      kind: Secret
+      name: $(.pod.spec.containers[?(@.name=="postgresql")].env[?(@.name=="POSTGRES_PASSWORD")].valueFrom.secretKeyRef.name)
```

Comment:
**suggestion** Perhaps a small comment with something like "The name of this input resource is dynamically read from another input resource`

Reply:
Also seems reasonable